tests/Integration/Behaviour/Features/Context/Domain/Carrier/CarrierFeatureContext.php

```diff
@@ -217,6 +224,18 @@ private function editCarrier(string $reference, ?string $newReference, TableNode
         if (isset($properties['group_access'])) {
             $command->setAssociatedGroupIds($this->referencesToIds($properties['group_access']));
         }
+
+        if (isset($properties['zones'])) {
+            $zones = $this->referencesToIds($properties['zones']);
+            $keyToFilter = 'id';
+        } else {
```

Comment:
This fallback should only be needed in the creation step (it avoids having to define all the parameters when we need to create a carrier quickly in other suites), that's also because you don't have this fallback in the creation step that so many errors are thrown right now.

The edition step doesn't need fallbacks, it simply handles partial update (only set the data in the command when it's explicitly set)